libfreerdp/core/tcp.c

```diff
@@ -1097,7 +901,9 @@ int freerdp_tcp_connect_multi(char** hostnames, UINT32* ports, int count, int po
 		}
 	}
 
-	waitStatus = WaitForMultipleObjects(count, events, FALSE, timeout * 1000);
+	events[count] = context->abortEvent;
+
+	waitStatus = WaitForMultipleObjects(count + 1, events, FALSE, timeout * 1000);
```

Comment:
I can see the WSAEventSelect,  it's in the diff ;). What I mean is the array 'events'.
Here's all lines of code involve it: 

        821 HANDLE* events;
	833 events = (HANDLE*) calloc(count + 1, sizeof(HANDLE));
        899 if (WSAEventSelect(sockfd, events[index], FD_READ | FD_WRITE | FD_CONNECT | FD_CLOSE))
      * 918 waitStatus = WaitForMultipleObjects(count + 1, events, FALSE, timeout * 1000);

We can see that the events in 'events' are all NULL.